### PR TITLE
Update 'SPIRV-Tools' to v2020.4-p0

### PR DESCRIPTION
**Pull request by zbai-sc**
* I've followed [this guide](https://hunter.readthedocs.io/en/latest/creating-new/update.html)
  step by step carefully. **Yes**

---

### CODE REVIEW

File: cmake/projects/SPIRV-Tools/hunter.cmake [hunk 1]
```diff
@@ -32,13 +32,36 @@ hunter_add_version(
...
+)
+
 hunter_cmake_args(
     SPIRV-Tools
     CMAKE_ARGS
       BUILD_TESTING=OFF
       SPIRV_SKIP_EXECUTABLES=ON
       SPIRV_SKIP_TESTS=ON
       SPIRV_WERROR=OFF
+      SPIRV_TOOLS_BUILD_STATIC=OFF
```
Comment (rbsheth) [inline cmake/projects/SPIRV-Tools/hunter.cmake:64]: This should be enabled/omitted to prefer static builds https://github.com/KhronosGroup/SPIRV-Tools/pull/3910

File: examples/SPIRV-Tools/config.cmake [hunk 2]
```diff
@@ -1,2 +1 @@
-hunter_config(SPIRV-Headers VERSION 1.5.1.corrected)
-
+hunter_config(SPIRV-Headers VERSION 1.5.4.raytracing.fixed)
```
Comment (rbsheth) [inline examples/SPIRV-Tools/config.cmake:1]: Nit: Please add a newline at the end

File: cmake/projects/SPIRV-Tools/hunter.cmake (hunk 1 above)
Comment (zbai-sc) [reply cmake/projects/SPIRV-Tools/hunter.cmake]: This is really complicated, so SPIRV_TOOLS_BUILD_STATIC won't actually change static/shared build type, that is still controlled by BUILD_SHARED_LIBS.
Only thing it changes is what ${SPIRV_TOOLS} will be aliased to.
Turning this on will break a lot of hunter changes regarding namespaces, I just got lazy and turned this off.

**Review by rbsheth — APPROVED** [review]
Mac toolchains with CMake >= 3.19 need the old buildsystem `-T buildsystem=1` for the Xcode generator. We should eventually fix this...

**Comment by rbsheth** [comment]
Released: https://github.com/cpp-pm/hunter/releases/tag/v0.23.294

**Pull request merged**